expression/builtin_miscellaneous.go

```diff
@@ -199,12 +199,14 @@ func (c *anyValueFunctionClass) getFunction(args []Expression, ctx context.Conte
 	case tpDuration:
 		sig = &builtinDurationAnyValueSig{baseDurationBuiltinFunc{bf}}
 	case tpInt:
+		bf.tp.Decimal = 0
 		sig = &builtinIntAnyValueSig{baseIntBuiltinFunc{bf}}
 	case tpJSON:
 		sig = &builtinJSONAnyValueSig{baseJSONBuiltinFunc{bf}}
 	case tpReal:
 		sig = &builtinRealAnyValueSig{baseRealBuiltinFunc{bf}}
 	case tpString:
+		bf.tp.Decimal = types.UnspecifiedLength
```

Comment:
@zz-jason We need more investigate about this. Known functions compatible with this rules are LOWER, UPPER, REVERSE, IF, ANYVALUE, ...

I guess that, when tpString is used as a parameter, its decimal is set to -1. I will add more examples about this behavior later.